ARCs/arc-0060.md

```diff
@@ -152,298 +182,206 @@ export interface HDWalletMetadata {
 export type Ed25519Pk = Uint8Array;
 ```
 
 > The wallet **MAY** want to operate directly with standard Algorand addresses. The transformation from a generic key to an Algorand address is left to the implementation. See <a href="https://developer.algorand.org/docs/get-details/accounts/#transformation-public-key-to-algorand-address">Public Key to Algorand Address</a> section of the developer portal.
 
-#### Interface `StdSigData`
-
-A `StdSigData` object represents a structured byte array of data to be signed by a wallet.
-
-```tsx
-export interface StdSigData {
-    /**
-    * The structured arbitrary data compliant with the JSON schema.
-    */
-    data: StdDataStr;
-    
-    /**
-    * The signer ed25519 public key.
-    */
-    signer: Ed25519Pk;
-
-    /**
-    * Optional HD wallet metadata used to derive
-    * the signer ed25519 public key.
-    */
-    hdPath?: HDWalletMetadata
-}
-```
-
-- `data` must be validated with the JSON schema in `metadata` (defined below).
-- `signer` is a `Ed25519Pk` object (defined below) that represents the signer public key.
-
 #### Interface `Signature`
 
 `Signature` is the produced 64-byte array, ed25519 digital signature of the signed message.
 
-```tsx
+```ts
 export type Signature = Uint8Array;
 ```
 
 #### Enum `ScopeType`
 
-The `ScopeType` enumerates constant strings with the scope of a signing action.
+The `ScopeType` enumerates constant strings with the scope of a signing action. Any extension of this ARC **SHOULD** adopt an existing `ScopeType`, or introduce a new one.
+
+This ARC introduces two scope types:
 
-This ARC introduces two scope types.
+| ScopeType          | Description |
+| -----------------  | ----------- |
+| MX_RANDOM          | Random data but it's prefixed with the domain separator `MX`. |
+| LSIG_TEMPLATE      | Signature of an Algorand program that compiles with a known template |
 
-| ScopeType | Description |
-| --- | --- |
-| AUTH | Signature of a basic 32 bytes challenge for authentication. |
-| LSIG | Signature of an Algorand program for delegation. |
+#### Error interface `SignDataError`
 
-For each `ScopeType` there is associated one JSON schema (see below).
+The `SignDataError` follows the same defined in [ARC-1](./arc-0001.md).
 
-Any extension of this ARC **SHOULD** adopt an existing `ScopeType`, or introduce a new one (with the respective JSON schema).
+```ts
+export class SignDataError extends Error {
+  code: number;
+  message: string;
+  data?: any;
+}
+```
 
-#### Interface `StdSignMetadata`
+`SignDataError` uses the same error codes of `SignTxnsError` as well as the following codes:
 
-A `StdSignMetadata` object specifies the metadata for the `StdSigData`.
+| Status Code | Name | Description |
+| --- | --- | --- |
+| 4600 | Invalid scope | The scope of the signing action is not recognized by the wallet |
+| 4601 | Doesn't match schema | The data does not adhere to the scope schema |
+| 4602 | Failed parsing | Data cannot be parsed according to the schema |
```

Comment:
What is the difference between `4600` and `4602`?

Reply:
Do you mean `4601` and `4602`? Good question if both are needed, probably not.